### PR TITLE
Creates a new baseline button for all of the app

### PR DESCRIPTION
**Pull request by orta**
Adds a new `Button` abstraction that should be used everywhere but the places where buttons need a progress indicator ( I think that's just inside the "follow artist" Card ). It's meant to be extended and then given theme settings which are then the buttons everyone uses.

I needed to show disabled states inside Consignments, and the fact that it needed wrapping to set a hight has been annoying me for a long time now: fixes https://github.com/artsy/emission/issues/544

I based the names on [reaction's names](https://artsy.github.io/reaction/?selectedKind=Components%2FButtons&selectedStory=Default%20Button&full=0&down=0&left=1&panelRight=0)

## TODO

- [x] Migrate consignments to this
- [x] Migrate the rest of the app

Skip new tests

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  No CHANGELOG added.
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>

  Missing Test Files:

- `src/lib/Components/Buttons/__tests__/Button-tests.tsx`
- `src/lib/Components/Buttons/__tests__/index-tests.tsx`

If these files are supposed to not exist, please update your PR body to include "Skip New Tests".
  </td>
    </tr>
  </tbody>
</table>




<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: src/data/colors.ts [hunk 1]
```diff
@@ -1,5 +1,21 @@
...
+  PurpleLight = "#e2d2ff",
+  PurpleRegular = "#6e1fff",
+  RedRegular = "#f7625a",
+  YellowRegular = "#fdefd1",
+  YellowMedium = "#fce1a8",
+  YellowBold = "#f1af1b",
+  GreenRegular = "#0eda83",
+  Black = "black",
+  White = "white",
+}
```
Comment (orta) [inline src/data/colors.ts]: I'm not going to migrate all of the app to this in this PR, but it's probably not too hard to migrate sometime

File: src/lib/Components/Buttons/Button.tsx [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+  /** Color setups for the different states */
+  stateColors: {
+    enabled: ColorTheme
+    highlighted: ColorTheme
+    disabled: ColorTheme
+  }
+}
+
+interface State {
+  /** What is  */
```
Comment (orta) [inline src/lib/Components/Buttons/Button.tsx]: I'll  add some more comments around these later

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  No CHANGELOG added.
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>

  Missing Test Files:

- `src/lib/Components/Buttons/__tests__/Button-tests.tsx`
- `src/lib/Components/Buttons/__tests__/index-tests.tsx`

If these files are supposed to not exist, please update your PR body to include "Skip New Tests".
  </td>
    </tr>
  </tbody>
</table>




<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Missing Test Files:

- `src/lib/Components/Buttons/__tests__/Button-tests.tsx`
- `src/lib/Components/Buttons/__tests__/index-tests.tsx`

If these files are supposed to not exist, please update your PR body to include "Skip New Tests".
  </td>
    </tr>
  </tbody>
</table>




<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Missing Test Files:

- `src/lib/Components/Buttons/__tests__/Button-tests.tsx`
- `src/lib/Components/Buttons/__tests__/index-tests.tsx`

If these files are supposed to not exist, please update your PR body to include "Skip New Tests".
  </td>
    </tr>
  </tbody>
</table>




<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Missing Test Files:

- `src/lib/Components/Buttons/__tests__/Button-tests.tsx`
- `src/lib/Components/Buttons/__tests__/index-tests.tsx`

If these files are supposed to not exist, please update your PR body to include "Skip New Tests".
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Missing Test Files:

- `src/lib/Components/Buttons/__tests__/Button-tests.tsx`
- `src/lib/Components/Buttons/__tests__/index-tests.tsx`

If these files are supposed to not exist, please update your PR body to include "Skip New Tests".
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: src/lib/Components/Consignments/Screens/Metadata.tsx [hunk 3]
```diff
@@ -52,10 +55,33 @@ const categoryOptions = [
 ]
 
 interface State extends ConsignmentMetadata {
-  showSelector?: boolean
+  showPicker?: boolean
+}
+
+// TODO: write a blog post about the refs
```
Comment (ashfurrow) [inline src/lib/Components/Consignments/Screens/Metadata.tsx]: 👍

File: src/lib/Components/Consignments/Screens/Overview.tsx [hunk 4]
```diff
@@ -153,7 +152,7 @@ export default class Info extends React.Component<Props, ConsignmentSetup> {
       state.metadata.title &&
       state.metadata.year
     )
-
+    console.log("Cansub:", canSubmit)
```
Comment (ashfurrow) [inline src/lib/Components/Consignments/Screens/Overview.tsx]: Intentional?

File: src/lib/Components/Consignments/Components/FormElements.tsx [hunk 5]
```diff
@@ -1,7 +1,7 @@
 import * as React from "react"
 
 import { ScrollView, TextProperties, View, ViewProperties, ViewStyle } from "react-native"
-import FlatWhite from "../../Buttons/FlatWhite"
```
Comment (ashfurrow) [inline src/lib/Components/Consignments/Components/FormElements.tsx]: pouring one out for my favourite class name.

File: src/lib/Components/Consignments/Screens/Overview.tsx (hunk 4 above)
Comment (orta) [reply src/lib/Components/Consignments/Screens/Overview.tsx]: No - thanks

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Missing Test Files:

- `src/lib/Components/Buttons/__tests__/Button-tests.tsx`
- `src/lib/Components/Buttons/__tests__/index-tests.tsx`

If these files are supposed to not exist, please update your PR body to include "Skip New Tests".
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Missing Test Files:

- `src/lib/Components/Buttons/__tests__/Button-tests.tsx`
- `src/lib/Components/Buttons/__tests__/index-tests.tsx`

If these files are supposed to not exist, please update your PR body to include "Skip New Tests".
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Missing Test Files:

- `src/lib/Components/Buttons/__tests__/Button-tests.tsx`
- `src/lib/Components/Buttons/__tests__/index-tests.tsx`

If these files are supposed to not exist, please update your PR body to include "Skip New Tests".
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Danger failed to run `dangerfile.ts`.
  </td>
    </tr>
  </tbody>
</table>



## Error Error
```
ENOENT: no such file or directory, open 'tslint-errors.json'
Error: ENOENT: no such file or directory, open 'tslint-errors.json'
    at Object.fs.openSync (fs.js:652:18)
    at Object.fs.readFileSync (fs.js:553:33)
    at Object.apply (/home/travis/build/artsy/emission/node_modules/vm2/lib/contextify.js:288:34)
    at Object.<anonymous> (/home/travis/build/artsy/emission/dangerfile.ts:112:34)
    at NodeVM.run (/home/travis/build/artsy/emission/node_modules/vm2/lib/main.js:424:27)
    at Object.<anonymous> (/home/travis/build/artsy/emission/node_modules/danger/distribution/runner/DangerfileRunner.js:139:24)
    at step (/home/travis/build/artsy/emission/node_modules/danger/distribution/runner/DangerfileRunner.js:40:23)
    at Object.next (/home/travis/build/artsy/emission/node_modules/danger/distribution/runner/DangerfileRunner.js:21:53)
    at /home/travis/build/artsy/emission/node_modules/danger/distribution/runner/DangerfileRunner.js:15:71
    at Promise (<anonymous>)
```
### Dangerfile
```
---------------------^
```
  
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  You have a `yarn run` inside the package.json. This is probably a mistake, see https://github.com/yarnpkg/yarn/issues/3595.
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Missing Test Files:

- `src/lib/Components/Buttons/__tests__/Button-tests.tsx`
- `src/lib/Components/Buttons/__tests__/index-tests.tsx`

If these files are supposed to not exist, please update your PR body to include "Skip New Tests".
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  You have a `yarn run` inside the package.json. This is probably a mistake, see https://github.com/yarnpkg/yarn/issues/3595.
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Missing Test Files:

- `src/lib/Components/Buttons/__tests__/Button-tests.tsx`
- `src/lib/Components/Buttons/__tests__/index-tests.tsx`

If these files are supposed to not exist, please update your PR body to include "Skip New Tests".
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Missing Test Files:

- `src/lib/Components/Buttons/__tests__/Button-tests.tsx`
- `src/lib/Components/Buttons/__tests__/index-tests.tsx`

If these files are supposed to not exist, please update your PR body to include "Skip New Tests".
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Missing Test Files:

- `src/lib/Components/Buttons/__tests__/Button-tests.tsx`
- `src/lib/Components/Buttons/__tests__/index-tests.tsx`

If these files are supposed to not exist, please update your PR body to include "Skip New Tests".
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Missing Test Files:

- `src/lib/Components/Buttons/__tests__/Button-tests.tsx`
- `src/lib/Components/Buttons/__tests__/index-tests.tsx`

If these files are supposed to not exist, please update your PR body to include "Skip New Tests".
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Missing Test Files:

- `src/lib/Components/Buttons/__tests__/Button-tests.tsx`
- `src/lib/Components/Buttons/__tests__/index-tests.tsx`

If these files are supposed to not exist, please update your PR body to include "Skip New Tests".
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Missing Test Files:

- `src/lib/Components/Buttons/__tests__/Button-tests.tsx`
- `src/lib/Components/Buttons/__tests__/index-tests.tsx`

If these files are supposed to not exist, please update your PR body to include "Skip New Tests".
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by orta** [comment]
I ditched the SVG snapshots for this PR, it's a shame, but it'll be trivial to bring it back sometime.

**Pull request merged**